packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts

```diff
@@ -4,20 +4,41 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {expect} from '@loopback/testlab';
+import axios, {AxiosProxyConfig, AxiosRequestConfig} from 'axios';
 import delay from 'delay';
 import http from 'http';
 import {AddressInfo} from 'net';
 import pEvent from 'p-event';
 import path from 'path';
-import makeRequest from 'request-promise-native';
 import rimrafCb from 'rimraf';
+import tunnel, {ProxyOptions as TunnelProxyOptions} from 'tunnel';
+import {URL} from 'url';
 import util from 'util';
 import {HttpCachingProxy, ProxyOptions} from '../../http-caching-proxy';
 
 const CACHE_DIR = path.join(__dirname, '.cache');
 
 const rimraf = util.promisify(rimrafCb);
 
+const axiosInstance = axios.create({
+  // Provide a custom function to control when Axios throws errors based on
+  // http status code. Please note that Axios creates a new error in such
+  // condition and the original low-level error is lost
+  validateStatus: () => true,
```

Comment:
Quoting from Axios docs:

```
  // `validateStatus` defines whether to resolve or reject the promise for a given
  // HTTP response status code. If `validateStatus` returns `true` (or is set to `null`
  // or `undefined`), the promise will be resolved; otherwise, the promise will be
  // rejected.
```

Have you considered using the simpler `null` variant?

```suggestion
  validateStatus: null,
```

Reply:
I prefer to make it explicit. Personally I find it confusing if `null` and `undefined` have different meanings in the config.